docs/guides/extensions/panels-create-items.md

```diff
@@ -0,0 +1,566 @@
+
+## Prepare the View
+
+Open the `panel.vue` file and you will see the starter template and script. Skip to the script section and import the
+following packages:
+
+```js
+import { useApi, useCollection, useStores } from '@directus/extensions-sdk';
+import { ref, watch } from 'vue';
+```
+
+In the `props`, `showHeader` is one of the built-in properties which you can use to alter your panel if a header is
+showing. Remove the text property and add the collection and fields properties as well as the width and height which is
+useful for styling:
+
+```js
+props: {
+	showHeader: {
+		type: Boolean,
+		default: false,
+	},
+	collection: {
+		type: String,
+		default: '',
+	},
+	fields: {
+		type: Array,
+		default: [],
+	},
+	response_format: {
+		type: String,
+		default: '',
+	},
+	width: String,
+	height: String,
+},
+```
+
+After the `props`, create a `setup(props)` section and create the variables needed:
+
+```js
+setup(props) {
+	const { useFieldsStore, usePermissionsStore } = useStores();
+	const fieldsStore = useFieldsStore();
+	const permissionsStore = usePermissionsStore();
+	const hasPermission = permissionsStore(props.collection, 'create');
+	const api = useApi();
+	const { primaryKeyField } = useCollection(props.collection);
+	const formData = ref({});
+	const fieldData = ref([]);
+
+	const formResponse = ref({});
+	const formError = ref({});
+	const responseDialog = ref(false);
+}
+```
+
+The `FieldsStore` fetches all of the collection’s fields, the `PermissionsStore` checks the current user’s access to the
+collection, and the `Collection` store for fetching information about the selected collection and the API for performing
+the final POST request.
+
+You will also need to capture a response to present to the user. The `response_format` contains a string where the user
+can create their own response with data from the API. A `v-dialog` can show an important message to the user. This
+requires a boolean value (here `responseDialoge`) to control the visibility of the dialog box.
+
+Create a `getFields` function to fetch the detailed information for each selected field then call the function
+afterwards so it populates the variable when the panel loads:
+
+```js
+function getFields() {
+	fieldData.value = [];
+	props.fields.forEach(field => {
+		fieldData.value.push(fieldsStore.getField(props.collection, field));
+	});
+}
+
+fetchData();
+```
+
+If the fields, collection, or response format is changed, the `getFields` function will need to be called again. Use the
+following code:
+
+```js
+watch(
+	[
+		() => props.collection,
+		() => props.fields,
+		() => props.response_format
+	],
+	() => {
+		getFields();
+	},
+);
+```
+
+Create a `submitForm` function. This will send the contents of `formData` to the selected collection and capture the
+response, resetting the form once successful. If an error occurs, the response is captured in the `formError` variable:
+
+```js
+function submitForm() {
```

Comment:
Opinion question: do we want to use async/await instead of .then().catch() for functions like this?

Reply:
No strong preference. Gonna resolve this, and come back to it when we end up rewriting all of these for secure extensions in the mid-term future.